### ISSUE TITLE
Dark UI Theme: Default font color, color of axis labels and color of axis tick labels seems to be the wrong color

### ISSUE THREAD

**Issue by aq-daniel** [nline/nline-plotlyjs-panel#3]
Hello,
Thank you very much for maintaining and updating the plotlyjs Panel. 
I came across the following problem, when using the dark UI theme in grafana: The color of all texts (font color) seems to default to some dark grey - which is really hard to read. In my opinion the default font color for the dark UI theme should be white - as white is also the default color for axis tick labels in standard grafana panels. 
Here is a screenshot as example. I marked the mentioned texts with red circles.

![grafik](https://user-images.githubusercontent.com/69959957/226335511-064ec2b3-6445-45f3-8e36-1b078557ba0d.png)

Also the gridlines in the screenshot seem to be white, whereas they are grey in standard grafana panels. Maybe the colors for fonts and gridlines got swapped somehow? In my opnion the default color for gridlines should be grey and the defaut font color should be white, when using the dark UI theme.

As a workaround I could of course set the colors in the 'Layout' section manually, but this would also effect the colors for the light UI theme and I need an acceptable visualisation of my plotlyjs Panel in both UI themes. Is there any solution to change the colors only for the dark UI theme? Or is there any other workaround to achieve this?

Thank you in advance for looking into this issue. If any further information is needed, feel free to ask, please.

Used Versions:
- Grafana v9.4.3
- nline-plotlyjs-panel v1.3.0

Kind regards

**Comment by jacksongoode** [comment]
@aq-daniel That's a good thought. Yes currently they are set here:
```
paper_bgcolor: rgba(0,0,0,0)
plot_bgcolor: rgba(0,0,0,0)
```
But that make the font blend with the background. I could just remove these so that the background could be a solid white which would prevent any visibility issues - but I like your idea of a dynamic background as a default. I could get the Grafana theme and then set the font color of the X and Y axis to white. That might work?

The proper solution would be for Plotly.js to have access to the theme templates they include in Python. https://plotly.com/python/templates/

**Comment by jacksongoode** [comment]
Do you think it might be viable to include Plotly's dark theme as a resource file to load here? I think that might be the best option. I can give it a shot or at least partially integrate it. Hopefully, they don't update this file too much.

**Comment by aq-daniel** [comment]
Yes, that should be a viable option in my opinion. Together with some kind of automatically loading or selecting this template file depending on the grafana UI theme. 
Personally, I think the default plotly themes don't get updates very often, so the template file shouldn't need a large amount of maintenance in the future.

Thank you very much for putting work into this. I'm looking forward to test the new feature as soon as it is available and provide you with some feedback.

**Comment by ldrame21** [comment]
Hello,
In other plugins like Apache Echart or HTML graphics, we have access to a GrafanaTheme variable named theme. We can then freely use theme.isDark to then set our dark theme as we want. It would be fantastic if you can add this functionality to the plugin. Here's the GrafanTheme class object: https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/themes/createV1Theme.ts

Thank you!

**Comment by MarcoPignati** [comment]
Hello, the issue described here is also an issue for us. We would also like to have support for changing theme dynamically, avoiding some panels to be manually edited. Is there an update that I missed?

**Comment by jacksongoode** [comment]
For now I would suggest making the background white and keeping the font black. This is an [unresolved request](https://github.com/plotly/plotly.js/issues/5519) with plotly.js not exposing theme selection (but doing so in the .py implementation).
```css
paper_bgcolor: rgba(0,0,0,0);
plot_bgcolor: rgba(0,0,0,0);
```

**Comment by mvalletta** [comment]
I would suggest this solution working with dark and light theme:
```js
let text_color = "black";
let paper_bgcolor = "white";

if (document.querySelector("body").classList.contains("theme-dark")) {
  text_color = "white";
  paper_bgcolor = "#11121701";
}

var layout = {
  paper_bgcolor: paper_bgcolor,
  font: {
    color: text_color,
  }
}
```

**Comment by jacksongoode** [comment]
Working on this now, will be apart of the next release.

**Comment by jacksongoode** [comment]
This is released in v2.0.0. It uses Grafana's color scheme and can be overridden of course.

**Comment by jacksongoode** [comment]
If other elements need to be swapped please let me know. I just did:
```js
    const textColor = theme.colors.text.primary;
    const backgroundColor = theme.colors.background.primary;
    const altbackgroundColor = theme.colors.background.secondary;

    const baseThemeLayout = {
      font: {
        color: textColor,
      },
      paper_bgcolor: backgroundColor,
      plot_bgcolor: backgroundColor,
      hoverlabel: {
        bgcolor: textColor,
      },
      xaxis: {
        gridcolor: altbackgroundColor,
      },
      yaxis: {
        gridcolor: altbackgroundColor,
      },
    };
 ```
